assets/js/modules/analytics/components/settings/OptionalSettingsView.js

```diff
@@ -0,0 +1,138 @@
+/**
+ * Optional Settings View component.
+ *
+ * Site Kit by Google, Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { __, _x } from '@wordpress/i18n';
+import { Fragment } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import DisplaySetting from '../../../../components/DisplaySetting';
+import { CORE_SITE } from '../../../../googlesitekit/datastore/site/constants';
+import { trackingExclusionLabels } from '../common/TrackingExclusionSwitches';
+import { MODULES_ANALYTICS } from '../../datastore/constants';
+const { useSelect } = Data;
+
+export default function OptionalSettingsView() {
+	const useSnippet = useSelect( ( select ) =>
+		select( MODULES_ANALYTICS ).getUseSnippet()
+	);
+	const canUseSnippet = useSelect( ( select ) =>
+		select( MODULES_ANALYTICS ).getCanUseSnippet()
+	);
+	const adsConversionID = useSelect( ( select ) =>
+		select( MODULES_ANALYTICS ).getAdsConversionID()
+	);
+	const anonymizeIP = useSelect( ( select ) =>
+		select( MODULES_ANALYTICS ).getAnonymizeIP()
+	);
+	const trackingDisabled =
+		useSelect( ( select ) =>
+			select( MODULES_ANALYTICS ).getTrackingDisabled()
+		) || [];
```

Comment:
Super-minor, but I think it makes more sense to modify the return value of the hook rather than do an `||` on the hook itself? Just because hooks need to run in the same order all the time without conditionals and—while this will always be run—it's nicer to leave any conditional execution/returns away from `useX()` calls 😅 

I know it was like this before and we probably do this elsewhere, but just something I think we can improve on 🙂 


```suggestion
		useSelect( ( select ) =>
			select( MODULES_ANALYTICS ).getTrackingDisabled() || []
		);
```